test/extended/crc/cmd/cmd.go

```diff
@@ -140,11 +140,14 @@ func CheckCRCStatus(state string) error {
 		expression = `.*CRC VM:.*\s.*: .*Stopped.*`
 	}
 
-	err := util.ExecuteCommand(CRC("status").ToString())
+	cmd := exec.Command("crc", "status")
+	out, err := cmd.Output()
```

Comment:
Because I need to explicitly get the stdout to check what was in it, and `util.ExecuteCommand` does not return that and isn't easily modifiable to do so (and even if it was, it's used in many many places which would need changing too). So going around it like above seemed better. I hope I am not overlooking something here, but that was the reasoning.

Reply:
looking at `util/shell.go` code, `CommandReturnShouldMatch("stdout"` calls `CompareExpectedWithActualContains(expected, shell.GetLastCmdOutput("stdout"))` which is supposed to get the command stdout output. Not clear why it's not working for you :-/

Reply:
I suggest keeping this one out of the PR so that we can merge it, and to address issues related to this in a follow-up PR.

Reply:
In integration tests, we do not use the shell construct (to simulate a user interacting with crc in one shell session). In e2e, we do (initialized [here](https://github.com/crc-org/crc/blob/02e1f697a0f8d8882e12684835a5ea1d683753fa/test/e2e/testsuite/testsuite.go#L157)): 
```go
err := util.StartHostShellInstance(testWithShell)
```
We had issues with running `crc setup` (with all the progress bars etc.) in e2e tests and had to move the execution of that command out of the general shell. So with integration, we didn't use the shell construct at all.

Given that we do not have an instance of [ShellInstance](https://github.com/crc-org/crc/blob/02e1f697a0f8d8882e12684835a5ea1d683753fa/test/extended/util/shell.go#L48) available, functions which rely on it won't work as expected. Therefore, instead of retrieving the stdout content from the `shell` variable that would contain it if it existed, we need to pass it between command execution (`crc status`) and output analysis (does it contain `Running`?) by hand. And that's what this commit does. Without it, `WaitForClusterInState` won't have any effect (except for wasting 10mins). 

So if we omit this change from this PR, integration tests will keep failing, albeit with a different error than initially when we started out.

Reply:
This would be a nice explanation to have in the commit log ;) 

`CheckCRCStatus` is used by both e2e and integration tests, is it ok to remove the use of `shell` for e2e?
And what about all the users of `ExecuteCommand()` in cmd.go? They also rely on `shell.go`, is this going to cause issues?

If this change is good enough to fix integration tests without breaking e2e, we can go with it, but some follow-up cleanups would be very useful imo (a clean separation between helpers using `exec.Command`, and helpers using `shell.go`). The former would only be used in integration tests, and the latter in e2e.

Reply:
> This would be a nice explanation to have in the commit log ;)

Will add it there.

> CheckCRCStatus is used by both e2e and integration tests, is it ok to remove the use of shell for e2e?

It is ok in the sense that it'll work. However, it'll dent the effort to simulate single-session user interaction with CRC.

> And what about all the users of ExecuteCommand() in cmd.go? They also rely on shell.go, is this going to cause issues?

As long as you don't need to retrieve anything aside from the error, it will do the job the same way.

> some follow-up cleanups would be very useful imo

I don't know the extent of this. It could have further consequences -- maybe it makes less sense to share the code between e2e and integration then. I opened an issue to track this: #3859

Reply:
> maybe it makes less sense to share the code between e2e and integration then

Maybe some go interface can be defined and used in the shared code. The interface would either make use of the shell code to run the command/get stdout/..., or would directly run the command for the integration tests.